password_security/tests/test_signup.py

```diff
@@ -0,0 +1,153 @@
+
+        # check created user
+        created_user = self.env["res.users"].search([("login", "=", "test_user")])
+        self.assertEqual(login, "test_user")
+        password_write_date = created_user.password_write_date
+        self.assertTrue(password_write_date)
+
+        # Weak password: ValidationError is raised
+        with self.assertRaises(ValidationError):
+            created_user.password = "test_user_password"
+        self.assertEqual(password_write_date, created_user.password_write_date)
+
+        # Stronger password: no error raised
+        created_user.password = "!asdQWE12345_3"
+        self.assertNotEqual(password_write_date, created_user.password_write_date)
+
+    def test_04_web_auth_signup_invalid_qcontext(self):
+        """It should catch AttributeError"""
+
+        with mock.patch("odoo.http.db_filter") as db_filter:
+            db_filter.side_effect = lambda dbs, host=None: [get_db_name()]
+            with self.assertRaises(AttributeError):
+                # 'TestPasswordSecuritySignup' object has no attribute 'session'
+                self.url_open(
+                    "/web/signup",
+                    data={
+                        "csrf_token": http.Request.csrf_token(self),
+                    },
+                )
+
+    def test_05_web_auth_signup_invalid_qcontext(self):
+        """It should catch EndTestException on signup qcontext"""
+        self.session = http.root.session_store.new()
+        self.opener = Opener(self.env.cr)
+        self.opener.cookies.set("session_id", self.session.sid, domain=HOST, path="/")
+
+        with mock.patch(
+            "odoo.addons.auth_signup.controllers.main.AuthSignupHome.get_auth_signup_qcontext"
+        ) as qcontext:
+            qcontext.side_effect = EndTestException
+            with self.assertRaises(HTTPError):
+                # Catch HTTPError: 400 Client Error: BAD REQUEST
+                self.signup("jackoneill", "!asdQWE12345_3")
+
+    def test_06_web_auth_signup_invalid_render(self):
+        """It should render & return signup form on invalid"""
+        self.session = http.root.session_store.new()
+        self.opener = Opener(self.env.cr)
+        self.opener.cookies.set("session_id", self.session.sid, domain=HOST, path="/")
+
+        with mock.patch("odoo.http.db_filter") as db_filter:
+            db_filter.side_effect = lambda dbs, host=None: [get_db_name()]
+            # Signup: no name or partner given for new user
+            response = self.url_open(
+                "/web/signup",
+                data={
+                    "login": "test",
+                    "csrf_token": http.Request.csrf_token(self),
+                },
+            )
+
+        # Ensure we stay in the signup page
+        self.assertEqual(response.request.path_url, "/web/signup")
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(
+            "Signup: no name or partner given for new user",
```

Comment:
Because of auth_signup_verify_email, the error message will be "That does not seem to be an email address" rather than "Signup: no name or partner given for new user". I would try with a "valid" login format like "test@test.com"

Reply:
Gaaaah auth_signup_verify_email just eats the error and throws a generic one. Maybe you could do some sort of check against the installed module? Or make a small PR to auth_signup_verify_email to add the thrown error to the error context.